src/Thinktecture.Relay.Server/Maintenance/StatisticsCleanupJob.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Options;
+using Thinktecture.Relay.Server.Persistence;
+
+namespace Thinktecture.Relay.Server.Maintenance
+{
+	/// <inheritdoc />
+	public class StatisticsCleanupJob : IMaintenanceJob
+	{
+		private readonly IStatisticsRepository _statisticsRepository;
+		private readonly StatisticsOptions _statisticsOptions;
+
+		/// <summary>
+		/// Initializes a new instance of the <see cref="StatisticsCleanupJob"/>.
+		/// </summary>
+		/// <param name="statisticsRepository">An instance of an <see cref="IStatisticsRepository"/>.</param>
+		/// <param name="statisticsOptions">An instance of an <see cref="IOptions{StatisticsOptions}"/>.</param>
+		public StatisticsCleanupJob(IStatisticsRepository statisticsRepository, IOptions<StatisticsOptions> statisticsOptions)
+		{
+			_statisticsRepository = statisticsRepository ?? throw new ArgumentNullException(nameof(statisticsRepository));
+			_statisticsOptions = statisticsOptions.Value ?? throw new ArgumentNullException(nameof(statisticsOptions));
+		}
+
+		/// <inheritdoc />
+		public async Task DoMaintenanceAsync(CancellationToken stoppingToken)
+		{
+			if (stoppingToken.IsCancellationRequested)
```

Comment:
Why is this necessary? Isn't it better to forward the token to the repository calls instead?

Reply:
implemented